src/librustc_typeck/check/_match.rs

```diff
@@ -867,6 +868,13 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
             self.check_pat_walk(&field.pat, field_ty, def_bm, true);
         }
 
+        // Require `..` if structure or variant has non_exhaustive attribute.
+        if adt.is_non_exhaustive() && !adt.did.is_local() && !etc {
```

Comment:
We are checking variant patterns here, but `adt.is_non_exhaustive()` is returns non-exhaustiveness of the variant's enum, not of the variant itself.
I.e. this will incorrectly generate an error.
```rust
// Crate 1
#[non_exhaustive]
enum E { V { a: u8 } }

// Crate 2
match ... {
    E::V { a } => {}
}
```
(Could you also add this case to tests?)

Reply:
Resolved.